src/richie/apps/search/indexers/courses.py

```diff
@@ -146,59 +157,126 @@ def build_es_query(request, facets):
                 start, end = value
                 queries = [
                     *queries,
-                    {
-                        "range": {
-                            param: {
-                                "gte": start.datetime if start else None,
-                                "lte": end.datetime if end else None,
+                    (
```

Comment:
Using dictionary or even better a nametuple will avoid calling query[0] and query[1] in the following.